test/concrete/flowErc1155/FlowTest.sol

```diff
@@ -298,4 +300,98 @@ contract Erc1155FlowTest is FlowERC1155Test {
             vm.stopPrank();
         }
     }
+
+    /// Should mint and burn tokens per flow in exchange for another token (e.g. ERC20).
+    function testFlowERC1155MintAndBurnTokensPerFlowForERC20Exchange(
+        uint256 erc20OutAmmount,
+        uint256 erc20InAmmount,
+        uint256 tokenId,
+        address alice,
+        string memory uri
+    ) external {
+        vm.assume(sentinel != erc20OutAmmount);
+        vm.assume(sentinel != erc20InAmmount);
+        vm.assume(sentinel != tokenId);
+        vm.assume(address(0) != alice);
+        vm.assume(erc20InAmmount > erc20OutAmmount);
+        vm.assume(!alice.isContract());
+
+        (IFlowERC1155V5 flowErc1155, EvaluableV2 memory evaluable) = deployIFlowERC1155V5(uri);
+        assumeEtchable(alice, address(flowErc1155));
+
+        {
+            vm.mockCall(address(iTokenA), abi.encodeWithSelector(IERC20.transferFrom.selector), abi.encode(true));
+            vm.mockCall(address(iTokenB), abi.encodeWithSelector(IERC20.transfer.selector), abi.encode(true));
+
+            vm.expectCall(
+                address(iTokenA),
+                abi.encodeWithSelector(IERC20.transferFrom.selector, alice, flowErc1155, erc20InAmmount),
+                2
+            );
+            vm.expectCall(address(iTokenB), abi.encodeWithSelector(IERC20.transfer.selector, alice, erc20OutAmmount), 2);
+        }
+
+        // Stack mint
+        {
+            ERC20Transfer[] memory erc20Transfers = new ERC20Transfer[](2);
+
+            erc20Transfers[0] =
+                ERC20Transfer({token: address(iTokenA), from: alice, to: address(flowErc1155), amount: erc20InAmmount});
+
+            erc20Transfers[1] =
+                ERC20Transfer({token: address(iTokenB), from: address(flowErc1155), to: alice, amount: erc20OutAmmount});
+
+            ERC1155SupplyChange[] memory mints = new ERC1155SupplyChange[](1);
+            mints[0] = ERC1155SupplyChange({account: alice, id: tokenId, amount: erc20InAmmount});
+
+            uint256[] memory stack = generateFlowStack(
+                FlowERC1155IOV1(
+                    mints,
+                    new ERC1155SupplyChange[](0),
+                    FlowTransferV1(erc20Transfers, new ERC721Transfer[](0), new ERC1155Transfer[](0))
+                )
+            );
+            interpreterEval2MockCall(stack, new uint256[](0));
+        }
+
+        {
+            vm.startPrank(alice);
+            flowErc1155.flow(evaluable, new uint256[](0), new SignedContextV1[](0));
+            vm.stopPrank();
+
+            assertEq(IERC1155(address(flowErc1155)).balanceOf(alice, tokenId), erc20InAmmount);
```

Comment:
the comment for this test says "in exchange for" but i only see an assertion for one balance change

Reply:
I wanted to check erc20 balances but we use mocked Erc20 tokens, so i guess i cannot check balances on that? 
What additional checks can i add?